test/tls-provider.c

```diff
@@ -839,18 +3537,75 @@ int tls_provider_init(const OSSL_CORE_HANDLE *handle,
                       void **provctx)
 {
     OSSL_LIB_CTX *libctx = OSSL_LIB_CTX_new();
+    OSSL_FUNC_core_obj_create_fn *c_obj_create= NULL;
+    OSSL_FUNC_core_obj_add_sigid_fn *c_obj_add_sigid= NULL;
+    OSSL_PROVIDER *defaultnullprov = NULL;
+    PROV_XOR_CTX *prov_ctx = NULL;
+
+    /* Fake signatures use HMAC and thus need default provider in NULL context */
+    if (!OSSL_PROVIDER_available(NULL, "default")) {
+        defaultnullprov = OSSL_PROVIDER_load(NULL, "default");
```

Comment:
This looks really suspect. We definitely should not be doing this since some of our tests rely on default *not* being available in the default libctx.

Reply:
Removed in https://github.com/openssl/openssl/pull/19312/commits/3cde11228ad9030f4d7efee78b60dd5bbea0ef0e : (HMAC replaced by EVP_Q_mac)